src/routes/opendata.js

```diff
@@ -0,0 +1,132 @@
+const OpenDataFacultyRepository = require("../repository/conUOpenDataFacultyRepository");
+const OpenDataCourseRepository = require("../repository/conUOpenDataCourseRepository");
+const OpenDataImportantDateRepository = require("../repository/conUOpenDataImportantDateRepository");
+const router = require('express').Router();
+
+
+/** runs once on server start then refreshes open data once every 24 hours (if --odrefreshinterval param is not specified;
+ * else refreshes every 'odrefreshinterval' ms); open data refresh is not run if --odrefresh param is not set to true
+ */
+if (process.env.npm_config_odrefresh === "true")
+    setInterval(
+        function openDataRefresh() {
+            OpenDataFacultyRepository.refreshFacultyData();
+            OpenDataCourseRepository.refreshCourseData();
+            OpenDataImportantDateRepository.refreshImportantDateData();
+
+            return openDataRefresh;
+        }(), (process.env.npm_config_odrefreshinterval && Number.isInteger(Number(process.env.npm_config_odrefreshinterval)) ? Number(process.env.npm_config_odrefreshinterval) : 86400000));
```

Comment:
we might not want to compute every milliseconds within a day ....

Reply:
Updated --odrefreshinterval to accept days instead of milliseconds; currently limited to 24 days max, defaults to 1 day otherwise. This is because setInterval accepts a 32-bit int, which limits the value to the equivalent of 24.855 days. Overflows otherwise. Might implement [this](https://catonmat.net/settimeout-setinterval) at a later date.